c7n/resolver.py

```diff
@@ -134,6 +137,15 @@ def __init__(self, data, manager):
         self.cache = manager._cache or NullCache({})
         self.resolver = URIResolver(manager.session_factory, self.cache)
 
+    def _add_api_key(self, headers):
```

Comment:
Could you please add a bunch of tests for this method (e.g., to check that it works as intended if api secret is set; it does right things if anything goes wrong, etc.)?